test/tools/router_check/router.cc

```diff
@@ -76,43 +71,31 @@ bool RouterCheckTool::compareEntriesInJson(const std::string& expected_route_jso
 
-    const std::unordered_map<std::string, std::function<bool(ToolConfig&, const std::string&)>>
-        checkers = {
-            {"cluster_name",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareCluster(tool_config, expected);
-             }},
-            {"virtual_cluster_name",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareVirtualCluster(tool_config, expected);
-             }},
-            {"virtual_host_name",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareVirtualHost(tool_config, expected);
-             }},
-            {"path_rewrite",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareRewritePath(tool_config, expected);
-             }},
-            {"host_rewrite",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareRewriteHost(tool_config, expected);
-             }},
-            {"path_redirect",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareRedirectPath(tool_config, expected);
-             }},
-        };
-
-    // Call appropriate function for each match case
-    for (std::pair<std::string, std::function<bool(ToolConfig&, std::string)>> test : checkers) {
+    using checkerFunc = std::function<bool(ToolConfig&, const std::string&)>;
+    const std::unordered_map<std::string, checkerFunc> checkers = {
+        {"cluster_name",
+         [this](auto&... params) -> bool { return this->compareCluster(params...); }},
+        {"virtual_cluster_name",
+         [this](auto&... params) -> bool { return this->compareVirtualCluster(params...); }},
+        {"virtual_host_name",
+         [this](auto&... params) -> bool { return this->compareVirtualHost(params...); }},
+        {"path_rewrite",
+         [this](auto&... params) -> bool { return this->compareRewritePath(params...); }},
+        {"host_rewrite",
+         [this](auto&... params) -> bool { return this->compareRewriteHost(params...); }},
+        {"path_redirect",
+         [this](auto&... params) -> bool { return this->compareRedirectPath(params...); }},
+    };
+
+    // Call appropriate function for each match case.
+    for (const auto& test : checkers) {
       if (validate->hasObject(test.first)) {
-        std::string expected = validate->getString(test.first);
+        const std::string& expected = validate->getString(test.first);
         if (tool_config.route_ == nullptr) {
           compareResults("", expected, test.first);
         } else {
-          if (!test.second(tool_config, validate->getString(test.first))) {
-            no_failures = false;
+          if (!test.second(tool_config, expected)) {
+            return false;
```

Comment:
Was there a reason to change the behavior from keep going to fail on first error?

Reply:
This has no effect on the `--details` flag. All it's used for in that function is to just print the test names:
```
    std::string test_name = check_config->getString("test_name", "");
    if (details_) {
      std::cout << test_name << std::endl;
    }
```

Otherwise, the tools relies on return values.

Reply:
There's one more place the details flag is used to control output: https://github.com/envoyproxy/envoy/blob/master/test/tools/router_check/router.cc#L228

Reply:
Ah, good catch! Put it back.